terraform/alb/alb.tf

```diff
@@ -0,0 +1,54 @@
+resource "aws_lb" "lb" {
+  name               = "${var.projectname}-lb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [data.aws_security_group.sg.id]
+  subnets            = data.aws_subnet_ids.subs.ids
+  tags = {
+    Name = "${var.projectname}-lb"
+  }
+}
+resource "aws_lb_listener" "lblistener" {
+  load_balancer_arn = aws_lb.lb.arn
+  port              = "80"
+  protocol          = "HTTP"
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.lbtg.arn
+  }
+}
+resource "aws_lb_target_group" "lbtg" {
+  name     = "${var.projectname}-lb-tg"
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = data.aws_vpc.vpc.id
+}
+resource "aws_lb_target_group_attachment" "lbtga" {
+  target_group_arn = aws_lb_target_group.lbtg.arn
```

Comment:
we need to link the ECS service with the load balancer target group, not the instances.. will send more instructions

Reply:
For this part, how do we link the alb with the ecs services, is it gonna be linked by the security group? which will be adding the sg of the alb to the sg of the ecs-ec2? I am modifying the ecs right now, and I create a security group in ecs, so i am expecting once the load balancer been created, and I can change this block, not sure if that's right approach `resource "aws_security_group" "ecs-ec2-sg" {
    name   = "${var.project_name}-ecs-ec2-sg"
    vpc_id = var.vpc_id

    lifecycle {
        create_before_destroy = true
    }

    # now make it possible getting from anywhere, after load balancer been created, it needs updated
    ingress {
        from_port       = 0
        to_port         = 0 #change to 65535 after loadbalance
        protocol        = -1 #change to tcp after loadbalancer
        cidr_blocks     = ["0.0.0.0/0"]
        #security_groups = [module.ecs_apps.ecs_nodes_secgrp_id] This will be ALB sg after ALB created
        description     = "From ALB"
    }
}`

Reply:
you need to link the target group with the ecs service
```
resource "aws_ecs_service" "default" {
  name                              = "${var.name}"
  cluster                           = "${var.cluster_name}"
  task_definition               = "${aws_ecs_task_definition.default.arn}"
  desired_count                = 1
  iam_role                          = "${var.service_role_arn}"

  load_balancer {
    target_group_arn = "${aws_lb_target_group.green.arn}"
    container_name   = "${var.name}"
    container_port   = "${var.container_port}"
  }
```